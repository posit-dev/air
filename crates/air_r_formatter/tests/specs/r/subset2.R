fn[[]]
fn[[a]]
fn[[a = 1, ... = 2]]

# Inherits call-like behavior with trailing braced expressions
fn[[a = { 1 + 1 }]]
fn[["description", {
  1 + 1
}]]

# Holes
fn[[,]]
fn[[,,]]
fn[[a,,b,,]]
fn[[a_really_long_argument_here,,another_really_really_long_argument_to_test_this_feature,,]]

# Dots
fn[[...]]
fn[[..., a = 1]]
fn[[a = 1, another_really_really_long_argument_to_test_this_feature, a_really_long_argument_here, ...]]

# Comments
fn[[
  # dangling special case
]]
