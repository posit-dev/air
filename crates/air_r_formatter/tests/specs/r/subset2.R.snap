---
source: crates/air_formatter_test/src/snapshot_builder.rs
info: r/subset2.R
---
# Input

```R
fn[[]]
fn[[a]]

# Inherits call-like behavior with trailing braced expressions
fn[[a = { 1 + 1 }]]
fn[["description", {
  1 + 1
}]]

# ------------------------------------------------------------------------
# Holes

# Leading holes should hug the `[[` token
fn[[,]]
fn[[,,]]

fn[[a,,b,,]]
fn[[a_really_long_argument_here,,another_really_really_long_argument_to_test_this_feature,,]]

# Holes are "invisible" when determining user requested expansion
# These all expand
fn[[,
  x = 1
]]
fn[[
  ,
  x = 1
]]
fn[[
  , x = 1
]]
fn[[
  ,, x = 1
]]

# ------------------------------------------------------------------------
# Dots

fn[[...]]
fn[[..., a = 1]]
fn[[a = 1, ... = 2]]
fn[[a = 1, another_really_really_long_argument_to_test_this_feature, a_really_long_argument_here, ...]]

# ------------------------------------------------------------------------
# Dot dot i

fn[[..1, ..2]]
fn[[..1 = 1, ..2 = 2]]

# ------------------------------------------------------------------------
# Comments

fn[[
  # dangling special case
]]

# ------------------------------------------------------------------------
# User requested line break

# A line break before the first argument forces expansion

# So this data dictionary stays expanded even though it fits on one line
df[[
  df$col > 1,
  c(2, 3)
]]

# This flattens to one line
df[[df$col > 1,
  c(2, 3)
]]

# This flattens to one line
df[[df$col > 1, c(2, 3)
]]

# Expanding the inner subset forces expansion of the outer subset
df[[df$col > 7, map[[
  names(df)
]]]]

# ------------------------------------------------------------------------
# Comments "after" holes

df[[,
  # comment
  x
]]

```


=============================

# Outputs

## Output 1

-----
Indent style: Space
Indent width: 2
Line ending: LF
Line width: 80
Persistent line breaks: Respect
Skip: None
-----

```R
fn[[]]
fn[[a]]

# Inherits call-like behavior with trailing braced expressions
fn[[a = {
  1 + 1
}]]
fn[["description", {
  1 + 1
}]]

# ------------------------------------------------------------------------
# Holes

# Leading holes should hug the `[[` token
fn[[,]]
fn[[,,]]

fn[[a, , b, , ]]
fn[[
  a_really_long_argument_here,
  ,
  another_really_really_long_argument_to_test_this_feature,
  ,
]]

# Holes are "invisible" when determining user requested expansion
# These all expand
fn[[,
  x = 1
]]
fn[[,
  x = 1
]]
fn[[,
  x = 1
]]
fn[[,,
  x = 1
]]

# ------------------------------------------------------------------------
# Dots

fn[[...]]
fn[[..., a = 1]]
fn[[a = 1, ... = 2]]
fn[[
  a = 1,
  another_really_really_long_argument_to_test_this_feature,
  a_really_long_argument_here,
  ...
]]

# ------------------------------------------------------------------------
# Dot dot i

fn[[..1, ..2]]
fn[[..1 = 1, ..2 = 2]]

# ------------------------------------------------------------------------
# Comments

fn[[
  # dangling special case
]]

# ------------------------------------------------------------------------
# User requested line break

# A line break before the first argument forces expansion

# So this data dictionary stays expanded even though it fits on one line
df[[
  df$col > 1,
  c(2, 3)
]]

# This flattens to one line
df[[df$col > 1, c(2, 3)]]

# This flattens to one line
df[[df$col > 1, c(2, 3)]]

# Expanding the inner subset forces expansion of the outer subset
df[[
  df$col > 7,
  map[[
    names(df)
  ]]
]]

# ------------------------------------------------------------------------
# Comments "after" holes

df[[,
  # comment
  x
]]
```
